src/lib/viz/style/helpers/icon-style.ts

```diff
@@ -0,0 +1,133 @@
+import { IconLayer } from '@deck.gl/layers';
+import { Feature, Point, MultiPoint } from 'geojson';
+import { GeometryType } from '@/viz/source';
+import { StyledLayer } from '../layer-style';
+import { Style, getStyles } from '..';
+import { CartoStylingError, stylingErrorTypes } from '../../errors/styling-error';
+
+export interface IconOptionsStyle {
+  // icon width in pixels
+  iconWidth: number;
+  // icon height in pixels
+  iconHeight: number;
+  // horizontal position of icon anchor
+  anchorX: number;
+  // vertical position of icon anchor
+  anchorY: number;
+  // Icon size multiplier
+  sizeScale: number;
+}
+
+interface IconOptionsStyleValidated {
```

Comment:
Why this 'Validated' suffix? Have we done something like this in other style helpers?

Reply:
TS things,it is the easiest solution to avoid errors